clients/localdir/client.go

```diff
@@ -92,22 +98,36 @@ func listFiles(clientPath string, predicate func(string) (bool, error)) ([]strin
 		}
 
 		// Remove prefix of the folder.
-		cleanPath := path.Clean(pathfn)
-		prefix := fmt.Sprintf("%s%s", clientPath, string(os.PathSeparator))
-		p := strings.TrimPrefix(cleanPath, prefix)
+		p := trimPrefix(pathfn, clientPath)
+		files = append(files, p)
+
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error walking the path %q: %w", clientPath, err)
+	}
+
+	return files, nil
+}
+
+func applyPredicate(clientPath string, clientFiles []string,
+	errFiles error, predicate func(string) (bool, error)) ([]string, error) {
+	if errFiles != nil {
+		return nil, errFiles
+	}
+
+	files := []string{}
+	for _, pathfn := range clientFiles {
+		// Remove prefix of the folder.
+		p := trimPrefix(pathfn, clientPath)
```

Comment:
Save the trimmed path to `client.files` to avoid running `trimPrefix` everytime?

Reply:
just realized the call was only needed once in the `do.Once`. My bad. So I removed the call entirely here.